### PR TITLE
Update redact method for sensitive json fields

### PR DESCRIPTION
**Pull request by dhavalpopat**
Currently we are logging private-key for one of the commands which is a security issue. This will redact any RunCommand parameter JSON value with "private-key" field. The solution can be extended to redact more keys as needed in the future.

Bug: https://msazure.visualstudio.com/One/_workitems/edit/26873623/

---

Testing:

Original Request
> mocctl.exe compute vm execute --name vm01 --group "mygroup" --command InstallArcAgent --params AgentConfig="{\"cloud\":\"AzureCloud\",\"private-key\":\"random+string/OutPut\"}"

Message without this PR change
> <name:"vm01-43450eba" id:"29C8FB3E-16D3-4B61-B1AD-E9429957F873" os:<> > Source:<CommandID:"InstallArcAgent" > RunCommandInputParameters:<Name:"AgentConfig" Value:"{\"cloud\":\"AzureCloud\",\"private-key\":\"random+string/OutPut\"}" >

Redacted Message with this PR change
> <name:"vm01-43450eba" id:"29C8FB3E-16D3-4B61-B1AD-E9429957F873" os:<> > Source:<CommandID:"InstallArcAgent" > RunCommandInputParameters:<Name:"AgentConfig" Value:"{\"cloud\":\"AzureCloud\",\"private-key\":\"** Redacted **\"}" >

### CODE REVIEW

**Comment by apluchik** [comment]
Could you please attach the output of what this would look like for a redacted String/non string field?

**Comment by dhavalpopat** [comment]
> Could you please attach the output of what this would look like for a redacted String/non string field?

Updated the description

**Review by apluchik — APPROVED** [review]

File: pkg/redact/redact.go [hunk 1]
```diff
@@ -72,7 +74,18 @@ func redactMessage(msg interface{}, val reflect.Value) {
 			}
 		}
 		if field.Options != nil {
-			ex, err := proto.GetExtension(field.Options, common.E_Sensitive)
+			ex, err := proto.GetExtension(field.Options, common.E_Sensitivejson)
+			if err != proto.ErrMissingExtension && (err != nil || *ex.(*bool)) {
```
[the remark this answers is not in the document]
Comment (dhavalpopat) [reply pkg/redact/redact.go:78]: Shouldn't it be `err == nil && *ex.(*bool)` ?

[the remark this answers is not in the document]
Comment (apluchik) [reply pkg/redact/redact.go]: yes! good catch, that should be correct logic

**Review by apluchik — APPROVED** [review]

**Pull request merged**